Save zoom state when writing stimulus parameters

bparks13 opened the pull request:
This PR now saves the zoom state when writing stimulus parameters, and reapplies the zoom state so that any changes can be immediately viewed without needing to zoom in again.

**Example GIF**
![save-zoom-state](https://github.com/user-attachments/assets/ff6b7f2d-e3f3-426c-b494-e711a7e2e4bb)

**Example GIF: Timebase Changes**
![timebase-change](https://github.com/user-attachments/assets/d824c5b2-94f8-44b6-94a2-5c8b6878e948)

Note that this may conflict with #459, since that PR refactors this class to be more generalized. Care needs to be taken that whichever one gets merged first is brought back into the other one so that the behavior remains correctly fixed.

Fixes #349

jonnew commented:
When I scroll aggressively, sometimes, all the waveforms disappear and I find them. This both should not happen and there should be a `Reset Zoom` button that brings everything back to default:

<img width="1188" height="684" alt="image" src="https://github.com/user-attachments/assets/9808d83b-36aa-4046-bacf-033ba6d713f2" />

jonnew commented:
Scale bars X/Y starting points need to account for line width so they dont look hacked

<img width="222" height="179" alt="image" src="https://github.com/user-attachments/assets/8afc80ed-38ae-4f82-a24b-8b64ad4ea7fe" />

jonnew commented on OpenEphys.Onix1.Design/Rhs2116StimulusSequenceDialog.cs:
```diff
@@ -247,12 +247,41 @@ private double GetPeakToPeakAmplitudeInMicroAmps()
+            public double XMin, XMax, YMin, YMax;
+        }
+
+        Zoom zoomState = null;
```
Do you need this global state? I see it used locally one time.

jonnew reviewed (changes requested):
I would start on this one and think about variable lifetimes a bit more.

bparks13 commented:
GIFs of behavior:

**Save Zoom State** [Note: no change since the last update in behavior, but implementation has been updated]

![rhs2116-zoom-state](https://github.com/user-attachments/assets/892dea8b-818f-451a-b664-cd7aed7af20b)

**Reset Zoom**

After removing the zoom-in boundary checking in https://github.com/open-ephys/bonsai-onix1/pull/497/commits/631c78fce9fc08ca3b95dde811071643eb294ba6, it is possible for users to zoom in excessively and lose all reference in the axis. The Reset Zoom button can get them out of this state now.

![reset-zoom](https://github.com/user-attachments/assets/87f697d4-9764-4854-9738-60506746e51d)

bparks13 commented on OpenEphys.Onix1.Design/Rhs2116StimulusSequenceDialog.cs:
```diff
@@ -262,12 +262,23 @@ private double GetPeakToPeakAmplitudeInMicroAmps()
             zedGraphWaveform.GraphPane.CurveList.Clear();
             zedGraphWaveform.GraphPane.GraphObjList.Clear();
+
+            var zoomState = new Zoom
```
Done, I have removed the zoom struct entirely.

jonnew reviewed (changes requested):
See my comment
